### PR TITLE
restyle the course home page

### LINKED ISSUE #2179: restyling for course home page
coursesfront

- [x] fix header text at top
- [x] make big red button text 'View All'
- [x] hide favorites if the user doesn't have any or if the user is anonymous (I think we can just hide whenever the response is empty)
- [x] fix styling for carousel titles
- [x] return to old 'previous' and 'next' button styles

### PR DESCRIPTION
**Pull request by aliceriot**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?

closes #2179

#### What's this PR do?

this does some styling tweaks to the course home page. also while I was fiddling with this component I refactored it to use the react-redux and redux-query hook APIs, and rewrote the tests to be tests of the full component.

#### How should this be manually tested?

make sure the styling fixes on the issue have all been satisfactorily implemented, and that it looks good and so on.

make sure all the code makes sense and everything.

#### Screenshots (if appropriate)

updatedcousehome

mobile is pretty sad rn but this is what it looks like:

updatedcousehomemob

### CODE REVIEW

**Comment by Ferdi** [comment]
If it's not too hard can you display on card in the mobile view ?

**Comment by aliceriot** [comment]
yeah - I haven't done any mobile stuff yet really but I can add that here

File: static/js/components/CourseCarousel.js [hunk 1]
```diff
@@ -16,50 +17,66 @@ type Props = {|
...
-  return disabled ? null : (
-    <div onClick={nextSlide} className="carousel-control next">
-      <i className="material-icons">chevron_right</i>
+  return (
+    <div className="course-carousel">
+      <div className="title-and-controls">
+        <div className="title">{title}</div>
+        <div className="controls">
+          <button
+            className={`dark-outlined compact extra-compact prev ${
```
Comment (mbertrand) [inline static/js/components/CourseCarousel.js]: What do you think of moving these buttons a bit closer to the carousel?  It can be a bit hard to tell which buttons belong to which carousel if you don't already know they should be on top:

Screen Shot 2019-09-11 at 11 31 09 AM

File: static/js/components/CourseCarousel.js [hunk 2]
```diff
@@ -16,50 +17,66 @@ type Props = {|
...
+            <i className="material-icons">arrow_forward</i>
+          </button>
+        </div>
+      </div>
+      <Carousel
+        slideIndex={index}
+        slidesToShow={CAROUSEL_PAGE_SIZE}
+        slidesToScroll="auto"
+        withoutControls={true}
+        afterSlide={slideIndex => setIndex(slideIndex)}
```
Comment (mbertrand) [inline static/js/components/CourseCarousel.js]: Is setting `afterSlide` necessary? The carousel seemed to behave the same with or without this line in some brief testing.

**Comment by aliceriot** [comment]
@ferdi ok I added a responsive thing to do 3 cards wide at desktop, 2 cards on a tablet, and 1 on a phone:

desksksksks
tabeltversion
phoneversion

File: static/js/factories/learning_resources.js [hunk 3]
```diff
@@ -133,3 +134,30 @@ export const makeLearningResource = (objectType: string) => {
...
+  const bootcamps = R.times(makeBootcamp, 3).map(makeFavorite)
+
+  return [
+    ...courses.map(formatFavorite(LR_TYPE_COURSE)),
+    ...programs.map(formatFavorite(LR_TYPE_PROGRAM)),
+    ...bootcamps.map(formatFavorite(LR_TYPE_BOOTCAMP))
+  ]
+}
+
+export const courseListResponse = (list: Array<LearningResource>) => ({
```
Comment (mbertrand) [inline static/js/factories/learning_resources.js]: This only seems to be used for tests, maybe it should go in `test_utils.js` instead?

File: static/js/components/CourseCarousel.js (hunk 1 above)
Comment (aliceriot) [reply static/js/components/CourseCarousel.js]: are you sure you pulled down the most recent version? the padding below the buttons should look like it does in the screenshots above

Comment (mbertrand) [reply static/js/components/CourseCarousel.js]: Just pulled again, looks good now 👍

File: static/js/components/CourseCarousel.js (hunk 2 above)
Comment (aliceriot) [reply static/js/components/CourseCarousel.js]: the `afterSlide` is what keeps the `slideIndex` in sync with the position of the carousel. The carousel has it's own internal state which manages which page it's showing - here basically I'm replicating that state externally (the `index` variable) so that I can control that state with the buttons. Then I pass that state down into the carousel (via the `slideIndex` prop) to control the carousel and sync and carousel-side changes to the position back to the `index` via this `afterSlide` callback.

basically, the case where it can get out of sync is if the user slides the carousel. if we're not syncing that change in the index back (via `afterSlide`) and the user then goes to click the button, we'll change teh `index` based on an incorrect current value for the index (b/c it hasn't been updated to reflect the changed position of the carousel) and we'll get some weird behavior.

does that make sense? I can show you how it can lead to weird behavior if that helps too

Comment (mbertrand) [reply static/js/components/CourseCarousel.js]: sounds good

File: static/js/factories/learning_resources.js (hunk 3 above)
Comment (aliceriot) [reply static/js/factories/learning_resources.js]: sure
